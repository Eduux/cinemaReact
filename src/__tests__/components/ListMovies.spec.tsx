import React from 'react';
import { render } from '@testing-library/react';

import Component from 'components/ListMovies';

describe('ListMovies component', () => {
  const list = [
    {
      imdbID: '1',
      poster: '1',
      title: 'filme',
      type: 'a',
      year: '112',
    },
  ];

  it('should render without data', async () => {
    const result = render(<Component list={[]} />);
    expect(result).toMatchSnapshot();
  });

  it('should render with data', async () => {
    const result = render(<Component list={list} />);
    expect(result).toMatchSnapshot();
  });

  it('should render without data and changing the empty list message', async () => {
    const result = render(<Component list={[]} emptyMessage="Teste" />);
    expect(result).toMatchSnapshot();
  });
});
