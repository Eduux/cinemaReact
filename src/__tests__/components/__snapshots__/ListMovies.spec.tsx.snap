// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMovies component should render with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-iCfLBT cscthS"
      >
        <li
          class="sc-dkPtyc kIhNcZ"
        >
          <div
            class="sc-eCImvq dyuREx"
          >
            <img
              alt="filme"
              class="sc-bdvvaa cLlfML"
              height="auto"
              loading="lazy"
              src="1"
              width="auto"
            />
          </div>
          <div
            class="sc-hKwCoD fRRmAz"
          >
            <strong
              class="sc-jRQAMF dxQwMq"
            >
              filme
            </strong>
            <strong
              class="sc-jRQAMF dxQwMq"
            >
              a
            </strong>
            <strong
              class="sc-jRQAMF dxQwMq"
            >
              112
            </strong>
            <div
              class="sc-gKckTs eMhRTy"
            >
              <button
                class="sc-gsDJrp ClvND"
                type="submit"
              >
                Favoritar
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-iCfLBT cscthS"
    >
      <li
        class="sc-dkPtyc kIhNcZ"
      >
        <div
          class="sc-eCImvq dyuREx"
        >
          <img
            alt="filme"
            class="sc-bdvvaa cLlfML"
            height="auto"
            loading="lazy"
            src="1"
            width="auto"
          />
        </div>
        <div
          class="sc-hKwCoD fRRmAz"
        >
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            filme
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            a
          </strong>
          <strong
            class="sc-jRQAMF dxQwMq"
          >
            112
          </strong>
          <div
            class="sc-gKckTs eMhRTy"
          >
            <button
              class="sc-gsDJrp ClvND"
              type="submit"
            >
              Favoritar
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListMovies component should render without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="sc-furvIG dnKQvo"
      >
        Nenhum filme encontrado ou pesquisado.
      </h2>
    </div>
  </body>,
  "container": <div>
    <h2
      class="sc-furvIG dnKQvo"
    >
      Nenhum filme encontrado ou pesquisado.
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListMovies component should render without data and changing the empty list message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="sc-furvIG dnKQvo"
      >
        Teste
      </h2>
    </div>
  </body>,
  "container": <div>
    <h2
      class="sc-furvIG dnKQvo"
    >
      Teste
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
